README.md

```diff
@@ -127,92 +126,93 @@ OakMap<K,V> oak = builder.build();
 
 ### OakMap Methods
 You are welcome to take a look on the OakMap's [full API](https://github.com/yahoo/Oak/wiki/Full-API).
-OakMap's API is similar to the ConcurrentNavigableMap API; some non-standard API methods and special cases are discussed below.
+OakMap's API implements the ConcurrentNavigableMap interface. For improved performance, it offers additional non-standard zero-copy API methods that are discussed below.
 
-### OakBuffers
-OakMap provides two types of memory buffers: *OakRBuffer* (read-only) and *OakWBuffer* (read and write). These buffers support the standard Java API for read-only Java ByteBuffers and writable Java ByteBuffers, respectively.
+### OakRBuffers
+OakMap provides read-only memory buffers, *OakRBuffer*` that support the standard Java API for read-only Java ByteBuffers.
 
 OakMap buffers allow the user direct access to the underlying serialized key-value pairs, without needing to worry about  concurrent accesses and memory management. This access reduces unnecessary copies and deserialization of the underlying mappings.
 Note, however, that since OakMap's get method avoids copying the value and instead returns access to the same underlying memory buffer that compute operations update in-place, the reader may encounter different values -- and even value deletions -- when accessing the buffer returned from get multiple times. This is of course normal behavior for a concurrent map that avoids copying.
 
-An OakRBuffer can represent either a key or a value. The OakRBuffer's user can use the standard interface of a *read-only* ByteBuffer, for example, `int getInt(int index)`, `char getChar(int index)`, `limit()`, etc. Notice that ConcurrentModificationException can be thrown as a result of any OakRBuffer method in case the mapping is concurrently deleted.
+The OakRBuffer's user can use the standard interface of a *read-only* ByteBuffer, for example, `int getInt(int index)`, `char getChar(int index)`, `limit()`, etc. Note that ConcurrentModificationException can be thrown as a result of any OakRBuffer method in case the mapping is concurrently deleted.
+
+For backward compatibility with applications that are already based on the use of ByteBuffers, OakRBuffer provides the transform method that atomically applies a transformation to the underlying ByteBuffer. For a more comprehensive code example please refer to the [Usage](#usage) section. 
```

Comment:
If use of ByteBuffer then only read-only ByteBuffer. It is not OK to give the direct ByteBuffer to the user. It may end up accessing wrong memory...